lib/install/management/resource_pool.go

```diff
@@ -69,22 +66,23 @@ func (d *Dispatcher) createResourcePool(conf *config.VirtualContainerHostConfigS
 	return rp, nil
 }
 
+// setResources will modify the resourceAllocation spec with the user provided allocation info
 func setResources(spec *types.ResourceAllocationInfo, resource types.ResourceAllocationInfo) {
-	*spec = resource
-	if spec.Limit == 0 {
-		spec.Limit = -1
+	if resource.Limit != nil {
+		// if no limit is requested then set to -1 for unlimited
```

Comment:
What is the default limit? Do we still need this logic at all if Limit can be omitted?
Is this used primarily to avoid the need for nil checks on Limit elsewhere in the code?

Reply:
This func is applying the user provided information to the default configuration.  So the user could omit the limit and then it would be -1 (default) or if this a configure operation and it's omitted then the limit won't be changed...